inception-project/inception
# #908 - Cannot start unless external search is enabled

**What's in the PR**
- Exlude any `.*AutoConfiguration` classes from component scanning
- Switch Spring configuration for External Search stuff from basic component scanning to Spring auto-configuration
- Introduce an order into the auto-configuration of External Search such that we can properly enable/disable the functionality
- Added various JavaDoc comments
- Fix width of the repository dropdown on the external search page
- Move `ElasticSearchTest.properties` to `src/test/resources`

**How to test manually**
* Test starting INCEpTION without `external-search.enabled=true` and see it runs
* Test starting INCEpTION with `external-search.enabled=true` check if external search works

**Automatic testing**

**Documentation**
